From 7e1f6903e89852a57d0925a1d52762be64c77c1d Mon Sep 17 00:00:00 2001
From: Dhruv Dang <hi@dhruv.io>
Date: Mon, 13 Jul 2020 22:28:58 -0700
Subject: [PATCH] Fix remaining issues from using haskell-src-exts >= 1.21

There were some breaking value constructor changes to the `Asst l` type in haskell-src-exts in v1.21. This PR fixes the remaining issues raised in #530. Please run tests locally before merging to double check my work.
---
 src/HIndent/Pretty.hs | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/HIndent/Pretty.hs b/src/HIndent/Pretty.hs
index 46d8254..1e18473 100644
--- a/src/HIndent/Pretty.hs
+++ b/src/HIndent/Pretty.hs
@@ -979,17 +979,20 @@ instance Pretty Alt where
 instance Pretty Asst where
   prettyInternal x =
     case x of
-      ClassA _ name types -> spaced (pretty name : map pretty types)
-      i@InfixA {} -> pretty' i
       IParam _ name ty -> do
         pretty name
         write " :: "
         pretty ty
+      ParenA _ asst -> parens (pretty asst)
+#if MIN_VERSION_haskell_src_exts(1,21,0)
+      TypeA _ ty -> pretty ty
+#else
+      ClassA _ name types -> spaced (pretty name : map pretty types)
+      i@InfixA {} -> pretty' i
       EqualP _ a b -> do
         pretty a
         write " ~ "
         pretty b
-      ParenA _ asst -> parens (pretty asst)
       AppA _ name tys ->
         spaced (pretty name : map pretty tys)
       WildCardA _ name ->
@@ -998,6 +1001,7 @@ instance Pretty Asst where
           Just n -> do
             write "_"
             pretty n
+#endif
 
 instance Pretty BangType where
   prettyInternal x =
-- 
2.26.2

